ops/framework.py

```diff
@@ -640,7 +641,7 @@ def drop_snapshot(self, handle):
         """Discard a persistent snapshot."""
         self._storage.drop_snapshot(handle.path)
 
-    def observe(self, bound_event: BoundEvent, observer: types.MethodType):
+    def observe(self, bound_event: BoundEvent, observer: typing.Callable[[typing.Any], None]):
```

Comment:
At first, the typing.Any made me unhappy. But it's not really worse than "types.MethodType," in terms of being pretty generic.

I think that I'm okay with this change.

Reply:
Agreed. Initially I [tried](https://github.com/canonical/operator/pull/664/commits/5d4adf35883381776c96e3fba944042a4f3a8afa) `Callable[[EventBase], None]` with the intention that all derived types to work, but mypy didn't like that.